### PR TITLE
[change] Added error 404 page #55

### LINKED ISSUE #55: add an Error 404 page
I would like to add an error 404 page to the website

### PR DESCRIPTION
**Pull request by AdarshRawat1**
[change] Added error 404 page

### CODE REVIEW

File: css/404.css [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+}
+.box-area li:nth-child(6) {
+  left: 15%;
+  width: 110px;
+  height: 110px;
+  animation-delay: 3.5s;
+}
+@keyframes animate {
+  0% {
+    transform: translateY(0) rotate(0deg);
```
Comment (Aryamanz29) [inline css/404.css:68]: Please add this `CSS` unit to  https://github.com/openwisp/OpenWISP-Website/blob/master/.stylelintrc.json#L17

File: css/404.css [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    opacity: 1;
+  }
+  100% {
+    transform: translateY(-800px) rotate(360deg);
```
Comment (Aryamanz29) [inline css/404.css:72]: Same here

**Review by Aryamanz29 — CHANGES_REQUESTED** [review]
Please change your commit PR title and commit message:

Run this command:

```bash
git commit --amend
```
Change your commit message to:
```
[change] Added error 404 page #55
Closes #55
```
and then always double check by running `./run-qa-checks` script on your development environment.

**Review by Aryamanz29 — CHANGES_REQUESTED** [review]
PR title : [change] Added error 404 page #55

File: 404.html [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+                        <i class="github icon"></i> Github
+                    </a>
+                </div>
+                <p>&copy; 2008-
+                    <script>document.write(new Date().getFullYear())</script> OpenWISP and individual contributors.
+                </p>
+            </div>
+        </div>
+    </div>
+    </div>
```
Comment (Aryamanz29) [inline 404.html:110]: You can remove this tag (`/div`) since tags must be paired, i.e., no start tag `<div>`.

**Review by Aryamanz29 — CHANGES_REQUESTED** [review]
Looks good @AdarshRawat1! but there is one minor change you must make to pass the quality assurance check.
I found that in your commit message, you've only changed the commit title, not the commit message body, i.e, `Closes #issue_number` is still missing.

To change that you've to run the following command on terminal:

```bash
git commit --amend
```

and change your commit message to:
```
# This should be your commit title
[change] Added error 404 page #55

# This should be your commit message body
Closes #55
```

and then you can force push to re-write this commit
```
git push origin master -f
```

**Comment by Aryamanz29** [comment]
Hey @AdarshRawat1 If you're facing any issues with setting up this project and running QA checks script, you can[ follow README instructions](https://github.com/Aryamanz29/OpenWISP-Website/tree/add-readme) that I have recently added in https://github.com/openwisp/OpenWISP-Website/pull/67.

**Review by Aryamanz29 — DISMISSED** [review]
Thanks @AdarshRawat1! I have tested your changes locally, and QA checks also passed. It looks good to me!

Deferring merge to @nemesisdesign.

Screenshot from 2023-03-14 00-04-28

Screenshot from 2023-03-13 23-58-58

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
> Thanks @AdarshRawat1! I have tested your changes locally, and QA checks also passed. It looks good to me!
> 
> Deferring merge to @nemesisdesign.
> 
> Screenshot from 2023-03-14 00-04-28
> 
> Screenshot from 2023-03-13 23-58-58

Why is the background different from the rest of the site?
The look and feel must be consistent. We cannot design different colors every time we add a new page, that is not the best practice.

**Comment by AdarshRawat1** [comment]
> > Thanks @AdarshRawat1! I have tested your changes locally, and QA checks also passed. It looks good to me!
> > Deferring merge to @nemesisdesign.
> > Screenshot from 2023-03-14 00-04-28
> > Screenshot from 2023-03-13 23-58-58
> 
> Why is the background different from the rest of the site? The look and feel must be consistent. We cannot design different colors every time we add a new page, that is not the best practice.

Should I remove the animation feature or simply change the color scheme?

**Review by nemesisdesign — COMMENTED** [review]
> > > Thanks @AdarshRawat1! I have tested your changes locally, and QA checks also passed. It looks good to me!
> > > Deferring merge to @nemesisdesign.
> > > Screenshot from 2023-03-14 00-04-28
> > > Screenshot from 2023-03-13 23-58-58
> > 
> > 
> > Why is the background different from the rest of the site? The look and feel must be consistent. We cannot design different colors every time we add a new page, that is not the best practice.
> 
> Should I remove the animation feature or simply change the color scheme?

We need a VERY SIMPLE 404 page with a consistent look and feel as the rest of the website. Adopt the KISS principle.

**Comment by AdarshRawat1** [comment]
> > > > Thanks @AdarshRawat1! I have tested your changes locally, and QA checks also passed. It looks good to me!
> > > > Deferring merge to @nemesisdesign.
> > > > Screenshot from 2023-03-14 00-04-28
> > > > Screenshot from 2023-03-13 23-58-58
> > > 
> > > 
> > > Why is the background different from the rest of the site? The look and feel must be consistent. We cannot design different colors every time we add a new page, that is not the best practice.
> > 
> > 
> > Should I remove the animation feature or simply change the color scheme?
> 
> We need a VERY SIMPLE 404 page with a consistent look and feel as the rest of the website. Adopt the KISS principle.

I acknowledge your instructions and have carried out the required actions. Kindly review the changes and inform me if any further adjustments are necessary.

File: 404.html [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    <script src="js/semantic.min.js"></script>
+    <script defer src="js/iconify.min.js"></script>
+    <script defer src="js/showdown.min.js"></script>
+    <script src="js/scripts.js"></script>
+    <script defer src="js/feed.js"></script>
+    <script>
+        $(document).ready(function () {
+            createActivityFeed({ "posts": 5, "loadMore": false });
+        });
+    </script>
```
Comment (nemesisdesign) [inline 404.html:108]: this script section can be removed

File: 404.html [hunk 5]
```diff
@@ -0,0 +1,110 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
+    <link rel="icon" type="image/png" href="images\favicon.png" />
+    <link rel="stylesheet" href="css/style.css" type="text/css" media="screen">
+    <link rel="stylesheet" href="css/semantic.min.css" type="text/css">
+    <title>Error 404</title>
```
Comment (nemesisdesign) [inline 404.html:9]:
```suggestion
    <title>Page Not Found (404)</title>
```

File: 404.html [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </div>
+        <div align="center">
+            <div class="ui hidden divider"></div>
+            <div class="ui hidden divider"></div>
+            <h1>Error 404!<br></h1>
+            <h2>The page you are looking for doesn't exist.<br>
```
Comment (nemesisdesign) [inline 404.html:56]:
```suggestion
            <h2>The page you're looking for can't be found.</h2>
```

File: 404.html [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+                    </h1>
+                    <div class="right menu">
+                        <div class="menu-open item">
+                            <i class="sidebar icon"></i>
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </div>
+        <div align="center">
```
Comment (nemesisdesign) [inline 404.html:52]: you're not following the structure of the other pages, please see how other content pages have "<article>" and the classes they usea and follow the same structure.

File: 404.html [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+            <div class="ui hidden divider"></div>
+            <div class="ui hidden divider"></div>
```
Comment (nemesisdesign) [inline 404.html:54]: get rid of these

File: 404.html [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+            <h1>Error 404!<br></h1>
```
Comment (nemesisdesign) [inline 404.html:55]:
```suggestion
            <h1>Page Not Found (404)</h1>
```

File: 404.html [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+            <h2>The page you are looking for doesn't exist.<br>
+                <br>
+                <br>
```
Comment (nemesisdesign) [inline 404.html:58]: remove the extra BRs

File: 404.html [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+        </div>
+        <div align="center">
+            <div class="ui hidden divider"></div>
+            <div class="ui hidden divider"></div>
+            <h1>Error 404!<br></h1>
+            <h2>The page you are looking for doesn't exist.<br>
+                <br>
+                <br> 
+        </div>
+            <p class="buttons center">
```
Comment (nemesisdesign) [inline 404.html:60]: fix the indentation here

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
It's looking better but you need to strictly adhere to the same HTML structure used in other pages, which is not being done here and hence the margin between elements looks odd and you had to add extra dividers and `<br>` elements which is not the way to go. This page is not consistent with the rest of the website.

**Comment by AdarshRawat1** [comment]
Kindly review the changes and guide me if any further adjustments are necessary.

**Review by nemesisdesign — APPROVED** [review]
Thanks!

**Pull request merged**